### ISSUE TITLE
As a Registered User, I want to Take a branch of a workflow referenced in EarthCODE Cat on CDSE, so that I can develop it further to enhance the workflow for the purpose of my science on CDSE

### ISSUE THREAD

**Issue by JanssenBrm** [ESA-EarthCODE/EarthCODE#15]

**Comment by JanssenBrm** [comment]
@GarinSmith  - Can we assume that the workflow implementation is open-sourced and publicly available? 

In short, if the implementation of a UDP is publicly available on GitHub, users can create a branch of this UDP implementation, apply the changes and publish it as a new experiment/workflow. This last step depends on https://github.com/ESA-EarthCODE/EarthCODE/issues/7, but in general, the other parts of the process are covered. 

Could you confirm if this is expected for this ticket to be considered as done? Please feel free to share your insights on additional use cases that should be covered.

**Comment by JanssenBrm** [comment]
Comment from @GarinSmith 

@JanssenBrm @edobrowolska @dgiomo 
Yes, I think we assume that the "workflow implementation is open-sourced and publicly available". We can deal with more complex scenarios later.  I note that the current  OpenEO OGC API Record implemenation includes  "rel": "license" which can assist in managing this.

I think platforms can decide how "users can create a branch of this UDP implementation".  
I think we re-use the existing approach above (EarthCODE/issues/7) if a user wants to publish a new experiment.
Yes I think the done criteria above is correct:
Done Criteria :  Users can create a branch of a published UDP implementation, apply the changes and publish it as a new experiment/workflow.

**Comment by JanssenBrm** [comment]
Thank you for the feedback @GarinSmith - I believe we can set this ticket to done. However, I can show this in a short demo during the next meeting to see if it matches the expectations.

**Comment by GarinSmith** [comment]
@JanssenBrm - Thanks, that sounds great if this capability to take a branch of a workflow already exists.  
However this story is part of the user journey to show that this is possible after discovering a workflow in the catalogue as part of an end to end journey. 
I think would happen after https://github.com/ESA-EarthCODE/EarthCODE/issues/11 most likely if that makes sense?  
It might help for you to demo this capability so that we can consider the end to end journey with more clarity.

**Comment by JanssenBrm** [comment]
Indeed, from a user journey perspective, this ticket makes more sense after implementing the previous steps. I'll keep it open for now to leave the option open to add new acceptance criteria based on previous steps.

**Comment by JanssenBrm** [comment]
For additional recommendations on editing workflows, please refer to https://github.com/ESA-EarthCODE/EarthCODE/issues/11#issuecomment-2750409620. We suggest storing the actual Python implementation as a link in the EarthCODE record, as demonstrated here: https://github.com/ESA-EarthCODE/open-science-catalog-metadata-testing/blob/main/workflows/worldcereal-workflow/record.json#L124-L127. This approach will enable EarthCODE users to branch off the implementation and adapt the workflow.
